test/unit/config_report_importer_test.rb

```diff
@@ -1,6 +1,6 @@
 require 'test_helper'
 
-class ReportImporterTest < ActiveSupport::TestCase
+class ConfigReportImporterTest < ActiveSupport::TestCase
```

Comment:
Could you try splitting this file up to test the generic ReportImporter and the ConfigReportImporter separately?  Only test the Puppet-specifics in the latter, and see if you can test the ReportImporter in a more generic, non-Puppet way.